.github/workflows/pr-workflow.yml

```diff
@@ -123,11 +131,85 @@ jobs:
+        uses: actions/checkout@v2.3.4
+        with:
+          ref: ${{ github.sha }}
+      - name: api_integration_test
+        run: |
+          STACK_NAME=${{needs.deploy.outputs.STACK_NAME}}
+          SOLUTION_NAME=${{needs.build.outputs.SOLUTION_NAME}}
+          VERSION=${{needs.build.outputs.VERSION}}
+          RESOURCE_ID=$(aws cloudformation describe-stack-resources --stack-name $STACK_NAME --logical-resource-id GlueStack --query "StackResources[0].PhysicalResourceId")
+          echo $RESOURCE_ID
+          NESTED_STACK_ID=$(echo $RESOURCE_ID | cut -d "/" -f 2)
+          echo "$NESTED_STACK_ID"
+          AMC_GLUE_JOB_NAME=$(aws cloudformation describe-stack-resources --stack-name $NESTED_STACK_ID --logical-resource-id "AmcGlueJob" --query "StackResources[0].PhysicalResourceId")
+          AMC_GLUE_JOB_NAME=$(echo "$AMC_GLUE_JOB_NAME" | tr -d '"')
+          ARTIFACT_BUCKET=$(aws cloudformation describe-stack-resources --stack-name $STACK_NAME --logical-resource-id "ArtifactBucket" --query "StackResources[0].PhysicalResourceId")
+          ARTIFACT_BUCKET=$(echo $ARTIFACT_BUCKET | tr -d '"')
+          echo $ARTIFACT_BUCKET
+          ROLE_NAME=$(aws cloudformation describe-stack-resources --stack-name $STACK_NAME --logical-resource-id "AmcApiAccessRole" --query "StackResources[0].PhysicalResourceId")
+          ROLE_NAME=$(echo $ROLE_NAME | tr -d '"')
+          echo $ROLE_NAME
+          CURRENT_ASSUME_ROLE_POLICY1=$(aws iam get-role --role-name $ROLE_NAME --query "Role.AssumeRolePolicyDocument.Statement[0]")
+          CURRENT_ASSUME_ROLE_POLICY2=$(aws iam get-role --role-name $ROLE_NAME --query "Role.AssumeRolePolicyDocument.Statement[1]")
+          TEST_ACCOUNT_ID=$(aws sts get-caller-identity --query Account)
+          TEST_ACCOUNT_ID=$(echo $TEST_ACCOUNT_ID | tr -d '"')
+          TEST_USER_ARN="arn:aws:iam::$TEST_ACCOUNT_ID:user/githubBuilderBot"
+          TEST_USER_ASSUME_ROLE_POLICY='{"Effect":"Allow","Principal":{"AWS":"'"$TEST_USER_ARN"'"},"Action":"sts:AssumeRole"}'
+          echo "$TEST_USER_ASSUME_ROLE_POLICY"
+          TEST_IAM_USER_POLICY='{"Version":"2012-10-17","Statement":['$CURRENT_ASSUME_ROLE_POLICY1', '$CURRENT_ASSUME_ROLE_POLICY2', '$TEST_USER_ASSUME_ROLE_POLICY']}'
+          aws iam update-assume-role-policy --role-name $ROLE_NAME --policy-document "$TEST_IAM_USER_POLICY"
+          AMC_GLUE_JOB_ROLE_NAME=$(aws cloudformation describe-stack-resources --stack-name $NESTED_STACK_ID --logical-resource-id "AmcGlueJobRole" --query "StackResources[0].PhysicalResourceId")
+          AMC_GLUE_JOB_ROLE_NAME=$(echo $AMC_GLUE_JOB_ROLE_NAME | tr -d '"')
+          export AMC_GLUE_JOB_NAME=$AMC_GLUE_JOB_NAME
+          export AMC_API_ROLE_ARN="arn:aws:iam::$TEST_ACCOUNT_ID:role/$ROLE_NAME"
+          export TEST_S3_BUCKET_NAME=${{ secrets.DATA_BUCKET_NAME }}
+          export AMC_ENDPOINT_URL=${{ secrets.AMC_API_ENDPOINT }}
+          export TEST_S3_KEY_NAME='amc-data-integ-test-2kb.json'
```

Comment:
We shouldn't assume that this file exists. It would be better to generate test data on-the-fly. Make sure all the timestamps in the test data are 6 months old or newer, otherwise we might trigger tickets to the AMC on-call team regarding user uploads of old data.

Reply:
fixed